go/app/main.go

```diff
@@ -121,36 +126,55 @@ func addItem(c echo.Context) error {
 	}
 
 	//Insert the data into the database
-	statement, _ := database.Prepare("INSERT INTO `Items` (`id`, `name`, `category_id`, `image_filename`) VALUES (?, ?, ?, ?);")
+	statement, err := database.Prepare("INSERT INTO `Items` (`name`, `category_id`, `image_filename`) VALUES (?, ?, ?);")
 	if err != nil {
 		log.Fatal(err)
 	}
 	defer statement.Close()
 
-	//Getting the id corresponding to the last item
-	var itemID int
-	err = database.QueryRow("SELECT id FROM Items ORDER BY id DESC LIMIT 1").Scan(&itemID)
-	if err != nil {
-		log.Fatal(err)
-	}
-
 	//Getting the id corresponding to the category that was given
 	var categoryID int
 	err = database.QueryRow("SELECT id FROM Category WHERE name = ?", category).Scan(&categoryID)
 	if err != nil {
 		fmt.Println("This category does not exist")
-		log.Fatal(err)
+		addCategory(category)
+		err = database.QueryRow("SELECT id FROM Category WHERE name = ?", category).Scan(&categoryID)
+		if err != nil {
+			log.Fatal(err)
+		}
```

Comment:
Could you try to use sql.Result in L172?
This query checking id can be replaced with sql.Result.LastInsertId in addCategory and addCategory can return inserted category's id.
https://pkg.go.dev/database/sql#Result